### PR TITLE
Add a simple CBOR parser to make it easier to test the CBOR writer.

### PR DESCRIPTION
**Pull request by jyasskin**

### CODE REVIEW

File: go/webpack/cbor/cbor_reader.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+)
+
+// A Decoder helps callers iterate through a CBOR buffer, extracting item
+// headers and string bodies as they go.
+type Decoder struct {
+	full []byte
```
Comment (mrdewitt) [inline go/webpack/cbor/cbor_reader.go]: nit: not clear what "full" is at first glance.

Comment (jyasskin) [inline go/webpack/cbor/cbor_reader.go]: Yeah, it's the full buffer that's being decoded. Does this just need a comment, or would you rather some other name?

Comment (mrdewitt) [inline go/webpack/cbor/cbor_reader.go]: probably prefer something like cborBytes or cborBuffer but that might just be my verbose C++ history. I leave it up to you.

Comment (jyasskin) [inline go/webpack/cbor/cbor_reader.go]: Sure, done.

**Pull request merged**